webpack/assets/javascripts/react_app/components/bookmarks/form/index.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { reduxForm } from 'redux-form';
+import { connect } from 'react-redux';
+import Form from '../../common/forms/Form';
+import TextField from '../../common/forms/TextField';
+import { required, length } from 'redux-form-validators';
+import * as FormActions from '../../../redux/actions/common/forms';
+
+class BookmarkForm extends React.Component {
+  // if we got here, its already post client validations
+  submit({ name, query, publik }, dispatch, props) {
+    const { submitForm, url, controller } = props;
+    const values = { name, query, public: publik || false, controller };
+
+    return submitForm({url, values, item: 'Bookmark'});
+  }
+
+  render() {
+    const { handleSubmit, submitting, error, onCancel } = this.props;
+
+    return (
+      <Form
+        onSubmit={handleSubmit(this.submit)}
+        onCancel={onCancel}
+        disabled={submitting}
+        submitting={submitting}
+        error={error}
+      >
+        <TextField
+          name="name"
+          type="text"
+          required="true"
+          label={__('Name')}
```

Comment:
Could this use model human attribute names?

Reply:
I'm not sure tbh, what would be the difference?

Reply:
avoid duplication of strings which helps in those rare cases we change some names, human names are used also in rendering validation messages so it would be easy to forget about